src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -1594,7 +1594,8 @@ public void run()
          @Override
          public void onCursorChanged(CursorChangedEvent event)
          {
-            timer_.schedule(1000);
+            if (prefs_.restoreSourceDocumentCursorPosition().getValue())
```

Comment:
Should we add an onValueChanged to the pref so that a restart isn't required to get this timer running?

Reply:
This code gets run in response to cursor changed events, and that handler is always attached. That should ensure we don't require a restart to get the timers running again since each cursor change we re-check the UI preference and trigger the timer if needed.

Reply:
ah, I missed that. LGTM!